### PR TITLE
Render HTML / CSS to Texture

### PR DESCRIPTION
**Pull request by 0X-JonMichaelGalindo**
**Description**

A convenient wrapper to render HTML / CSS to a texture. Specifically for use in VR user interfaces, but might have other uses.

---

MrDoob, do you want this for the main branch?

I have been using something similar in a VR project, and I thought it might be convenient for other VR developers.
I formatted it to match three.js textures, since that seemed to make the most sense.
If you want it but would like me to implement it differently, let me know. (If you think it doesn't fit with core three.js, that's fine too, I could just make it a separate extension or something.)

HTMLTexture Preview

### CODE REVIEW

**Comment by Mugen87** [comment]
If this is really wanted, I think it's better to put it in `examples/jsm/textures` and not in the core.

Besides, have you tested your example? It does not work: 

https://raw.githack.com/0X-JonMichaelGalindo/three.js/dev/examples/webgl_html_texture.html

Also please resolve all formatting and linter issues in your code.

**Comment by 0X-JonMichaelGalindo** [comment]
You would have to rebuild three.js to include the new texture. But it definitely works. (The screen cap is from the example.)

Well, if no one wants it, I'll just drop it.

**Comment by Mugen87** [comment]
> You would have to rebuild three.js to include the new texture.

You can import it as a module from the examples directory. We also do this for more important modules like `GLTFLoader` and `OrbitControls`.

File: examples/webgl_html_texture.html [hunk 1]
```diff
@@ -0,0 +1,146 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>three.js webgl - geometry - cube</title>
```
Comment (Mugen87) [inline examples/webgl_html_texture.html:4]: This title is wrong.

File: examples/webgl_html_texture.html [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                    <span>HTML Texture</span>
+                    <div id="load-bar"><div id="load-line" style="width:0%;"></div></div>
+                </div>
+
+            </div>
+        </div>
+
+		<script type="module">
+
+			import * as THREE from '../build/three.module.HTMLTexture.js';
```
Comment (Mugen87) [inline examples/webgl_html_texture.html:81]: This should be:
```js
import * as THREE from '../build/three.module.js';
```

File: examples/webgl_html_texture.html [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			animate();
+
+			function init() {
+
+				camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 1, 1000 );
+				camera.position.z = 400;
+
+				scene = new THREE.Scene();
+
+                const html = document.getElementById( "texture-html" ).innerHTML;
```
Comment (Mugen87) [inline examples/webgl_html_texture.html:97]: Formatting issues.

File: examples/webgl_html_texture.html [hunk 4]
```diff
@@ -79,7 +79,8 @@
 		<script type="module">
 
 			import * as THREE from '../build/three.module.js';
-
+			import HTMLTexture from '../src/textures/HTMLTexture.js';
```
Comment (Mugen87) [inline examples/webgl_html_texture.html:82]: Actually it should be:
```js
import { HTMLTexture } from './jsm/textures/HTMLTexture.js';
```

**Comment by 0X-JonMichaelGalindo** [comment]
I added a rebuilt version of three.js so you can at least see the example running on Githacks now. (Code cannot be pulled now, but I'll probably just delete & redo anyway.)

If anyone wants it, I'll look into reformatting it into the examples folder instead. Otherwise, I need to get back to work.

Thanks for taking a look.

**Pull request closed without merging**